Release v0.1 🎉

VitalInnovator opened the pull request:
<img width="1329" alt="localhost:3000:dashboard:roszcz%2Fgiant-midi-susta" src="https://github.com/Nospoko/data-quality-platform/assets/136322674/73090f3f-6a3b-4277-881e-652eec494b26">

I found an issue with using too much memory when working with MIDI files. I plan to investigate this problem after fixing the other bugs.
<img width="933" alt="Activity Monitor 2023-09-13 07-47-52" src="https://github.com/Nospoko/data-quality-platform/assets/136322674/6ae8d14f-b7e5-4d76-8283-87f253b505d1">

### Steps to Run
1. **Database Structure**: Confirm your database has the correct table structure (e.g., the `record` table should not have a `metadata` field). Optionally, reset the database.

2. **Environment Variables**: Make sure to set `DATA_PROBLEM=ecg_classification` and a valid dataset like `DATASET_NAME=roszcz/qrs-swipe-demo`.

3. **If you have reset the database**:
    3.1. **API Timing**: Make sure the Python API starts after the migration completes. If not, restart the containers with `docker-compose up`.
    3.2. **Login**: Log into the application after the front-end is up.
    3.3. **User Role**: Directly in the database, change the user role to 'admin'.
    3.4. **Dataset Creation**: In the admin panel, create a dataset name that actually exists. Make it active and synchronize it with the API.
    3.5. **Organization**: Create an organization, give it access to the dataset, and add your user to this organization.

4. **Dashboard**: Navigate to the dashboard and select the dataset with the actual name.

Allowed DATA_PROBLEM
` 'ecg_classification',
  'midi_review',
  'ecg_segmentation',`

VitalInnovator committed:
chore: update next project structure

VitalInnovator committed:
chore: add DATA_PROBLEM check to app

VitalInnovator committed:
feat: rate + entity

VitalInnovator committed:
feat: add AddScoreFeedback migration

VitalInnovator committed:
feat: midi feedback api

VitalInnovator committed:
feat: midi player

VitalInnovator committed:
fix: fix visualizer

VitalInnovator committed:
feat: player styles

VitalInnovator committed:
fix: fix url

VitalInnovator committed:
fix: snap back on finish

VitalInnovator committed:
doc: add player.md

VitalInnovator committed:
Merge branch 'develop' into qrs-34

VitalInnovator committed:
fix: minor fixes after branch merge

roszcz committed:
QRS-45: Python API: serve MIDI dataset (#27)

* prototype serving midi files

* store midi records metadata in the database

* column unification tryouts

* prepare dataset loader for schema updates

VitalInnovator committed:
feat: update columns in records table

VitalInnovator committed:
fix: comment unused code

VitalInnovator committed:
temp hardcode datasetName

VitalInnovator committed:
feat: add ECG Segmentation ranges

VitalInnovator committed:
add more comments

VitalInnovator committed:
remove hardcoded dataset

VitalInnovator committed:
fix: remove unnecessary data problem checks

VitalInnovator committed:
feat: add modal for editing ECG segments ranges

VitalInnovator committed:
feat: add drag-n-drop for range boundaries

VitalInnovator committed:
feat: add healthcheck to web service

VitalInnovator committed:
add DATA_PROBLEM env var to next build-time

VitalInnovator committed:
feat: highlight range area on hover; update RangesModal

VitalInnovator committed:
feat: add "Delete" button when hovering over a range

VitalInnovator committed:
feat: show segmentation ranges in ZoomView

VitalInnovator committed:
fix: reset values when closing RangesModal without saving

VitalInnovator committed:
change range values format to 0-1200

VitalInnovator committed:
feat: add metadata field to data_checks

VitalInnovator committed:
Merge branch 'qrs-34/add-score' into qrs-46

VitalInnovator committed:
fix: fix table name

VitalInnovator committed:
fix: fix prod build

roszcz committed:
adjust dataset ingestion for the new db schema

VitalInnovator committed:
chore: combine migrations

VitalInnovator committed:
fix: metadata

VitalInnovator committed:
Merge branch 'qrs-34/add-score' into qrs-44

VitalInnovator committed:
py api update

VitalInnovator committed:
feat: segmentation problem display

VitalInnovator committed:
fix: minor fix

VitalInnovator committed:
fix: fix midi player

VitalInnovator committed:
Merge branch 'qrs-48' into feat/combined-branch_v2

VitalInnovator committed:
fix: comment test code

VitalInnovator committed:
fix: fix migration

roszcz commented:
@VitalInnovator When running the `ecg_classification` task, in the zoom model I get the widget for segmentation:

https://github.com/Nospoko/data-quality-platform/assets/8056825/eb910cc1-2e36-4528-a2f9-05ab73f1e00b

VitalInnovator committed:
fix: exam_uid queries

VitalInnovator committed:
fix: minor fixes

VitalInnovator committed:
impr: filter logic

VitalInnovator committed:
fix: fixes

VitalInnovator committed:
impr: segmentation flow

VitalInnovator committed:
fix: ui fixes

VitalInnovator committed:
impr: ranges modal in zoom view

VitalInnovator commented:
> When running the `ecg_classification` task, in the zoom model I get the widget for segmentation:

I'll push fix at next 30m

roszcz reviewed (changes requested):
@VitalInnovator looks that the app is using the `/data` endpoint, but it's supposed to be deprecated. You should use the `record_id` from the `records` table and the `/record` endpoint. Let me know if this makes sense!

VitalInnovator commented:
> @VitalInnovator looks that the app is using the `/data` endpoint, but it's supposed to be deprecated. You should use the `record_id` from the `records` table and the `/record` endpoint. Let me know if this makes sense!

Got it. I'll fix it

VitalInnovator committed:
fix: fix segment overlap

roszcz commented:
@VitalInnovator I just noticed that in the `ecg_classification`, the history page shows only 5 samples, and clicking "Load more" doesn't do anything. The same seems to be the case for the midi review

VitalInnovator commented:
@roszcz regarding ranges in metadata. FE use this format 
```
{
X1:  [520, 680],
X2:  [618, 884],
R: [320, 495],
...
}
```

roszcz commented:
> @roszcz regarding ranges in metadata. FE use this format
> 
> ```
> {
> X1:  [520, 680],
> X2:  [618, 884],
> R: [320, 495],
> ...
> }
> ```

@VitalInnovator Is it ok if I add a field `segments` to the metadata, where another json can be stored:

```json
{
    "exam_uid": "3e1b17f3-3c37-4c3f-a723-863097d68f3d",
    "label": "X",
    "position": 16339207,
    "segments": "{\"R\": [545, 674], \"P\": [200, 260], \"Q\": [369, 467], \"S\": [732, 810], \"T\": [864, 973]}",
    "time": "2023-07-13T18:49:40.812217"
}
```

After parsing the json you'll be able to get this:

```json
{
    "R": [545, 674],
    "P": [200, 260],
    "Q": [369, 467],
    "S": [732, 810],
    "T": [864, 973]
}
```

VitalInnovator commented:
> > @roszcz regarding ranges in metadata. FE use this format
> > ```
> > {
> > X1:  [520, 680],
> > X2:  [618, 884],
> > R: [320, 495],
> > ...
> > }
> > ```
> 
> @VitalInnovator Is it ok if I add a field `segments` to the metadata, where another json can be stored:
> 
> ```json
> {
>     "exam_uid": "3e1b17f3-3c37-4c3f-a723-863097d68f3d",
>     "label": "X",
>     "position": 16339207,
>     "segments": "{\"R\": [545, 674], \"P\": [200, 260], \"Q\": [369, 467], \"S\": [732, 810], \"T\": [864, 973]}",
>     "time": "2023-07-13T18:49:40.812217"
> }
> ```
> 
> After parsing the json you'll be able to get this:
> 
> ```json
> {
>     "R": [545, 674],
>     "P": [200, 260],
>     "Q": [369, 467],
>     "S": [732, 810],
>     "T": [864, 973]
> }
> ```

 Yes. it's ok

roszcz committed:
add default segments for ecg_segmentation

roszcz commented:
@VitalInnovator done!

VitalInnovator committed:
fix: fix date format

VitalInnovator committed:
fix: ranges naming

VitalInnovator committed:
fix: fix midi player

VitalInnovator committed:
fix: fix load more

VitalInnovator committed:
fix: minor fixes

VitalInnovator commented:
Fixed adding "X" ranges on history page (index now depends on existing ranges but component state)
Fixed delete range buttons in segmentation plugin
Fixed load more button on history page
Fixed ECG record description + record time formatting via date-fns
Fixed issue with MIDI player (now it stops playing on feedback submit)
Handled segments in metadata (segments in DB always represented as JSON string and parsed on API level)
Other minor fixes and improvemets
